src/test/java/com/commercetools/sync/services/impl/CartDiscountServiceImplTest.java

```diff
@@ -0,0 +1,119 @@
+
+class CartDiscountServiceImplTest {
+
+    private String testCartDiscountKey = "ABC123";
+
+    @Test
+    void fetchCartDiscount_WithEmptyKey_ShouldNotFetchAnyCartDiscount() {
+        // preparation
+        final SphereClient sphereClient = mock(SphereClient.class);
+        final CartDiscountSyncOptions cartDiscountSyncOptions = CartDiscountSyncOptionsBuilder
+                .of(sphereClient)
+                .build();
+        final CartDiscountServiceImpl cartDiscountService = new CartDiscountServiceImpl(cartDiscountSyncOptions);
+
+        // test
+        CompletionStage<Optional<CartDiscount>> result = cartDiscountService.fetchCartDiscount("");
+
+
+        // assertions
+        assertThat(result).isCompletedWithValue(Optional.empty());
+        verify(sphereClient, never()).execute(any());
+    }
+
+    @Test
+    void fetchCartDiscount_WithNullKey_ShouldNotFetchAnyCartDiscount() {
+        // preparation
+        final SphereClient sphereClient = mock(SphereClient.class);
+        final CartDiscountSyncOptions cartDiscountSyncOptions = CartDiscountSyncOptionsBuilder
+                .of(sphereClient)
+                .build();
+        final CartDiscountServiceImpl cartDiscountService = new CartDiscountServiceImpl(cartDiscountSyncOptions);
+
+        // test
+        CompletionStage<Optional<CartDiscount>> result = cartDiscountService.fetchCartDiscount(null);
+
+
+        // assertions
+        assertThat(result).isCompletedWithValue(Optional.empty());
+        verify(sphereClient, never()).execute(any());
+    }
+
+    @Test
+    void fetchCartDiscount_WithValidKey_ShouldReturnMockCartDiscount() {
+        // preparation
+        final SphereClient sphereClient = mock(SphereClient.class);
+        final CartDiscount mockCartDiscount = mock(CartDiscount.class);
+        when(mockCartDiscount.getId()).thenReturn("testId");
+        when(mockCartDiscount.getName()).thenReturn(LocalizedString.ofEnglish("eng"));
+        final CartDiscountSyncOptions cartDiscountSyncOptions = CartDiscountSyncOptionsBuilder
+                .of(sphereClient)
+                .build();
+        final CartDiscountServiceImpl cartDiscountService = new CartDiscountServiceImpl(cartDiscountSyncOptions);
+        final PagedQueryResultDsl pagedQueryResult = mock(PagedQueryResultDsl.class);
+        when(pagedQueryResult.head()).thenReturn(Optional.of(mockCartDiscount));
+        when(cartDiscountSyncOptions.getCtpClient().execute(any())).thenReturn(CompletableFuture
+                .completedFuture(pagedQueryResult));
+
+        // test
+        CompletionStage<Optional<CartDiscount>> result = cartDiscountService.fetchCartDiscount(testCartDiscountKey);
+
+
+        // assertions
+        assertThat(result).isCompletedWithValue(Optional.of(mockCartDiscount));
+        verify(sphereClient, only()).execute(any());
+    }
+
+    @Test
+    void createCartDiscount_WithEmptyCartDiscountKey_ShouldNotCreateCartDiscount() {
```

Comment:
Where are the other tests like update cart discount, successfully create cart discount etc.?

Reply:
here the tests from Jude, some of the service tests are already covered by IT's but I will check what could be added to here.

Reply:
I forgot to add this before, here I try to add some unit tests with mock responses from CTP,  80c2eed